test/integration/push-jenkins-update.test.js

```diff
@@ -19,7 +19,10 @@ tap.test('Sends POST requests to https://api.github.com/repos/nodejs/node/status
                   .reply(201)
 
   t.plan(1)
-  t.tearDown(() => prCommitsScope.done() && scope.done())
```

Comment:
The previous approach of testing if network requests had been sent from the bot, via nock scopes, didn't test more then one network request.

That's because we tested all scoped in a one-liner like the above.

It turns out that the above will not invoke the second `scope.done()` (because `.done() returns `undefined`), meaning only the first nock scope (read: outgoing network request) was asserted. By going away from that one-liner and invoking one by one on each line, tests starts to explode.

Reply:
Why do we have to wait for the `tearDown`?

Reply:
That's a great question!

I don't have a good answer to why it was placed there in the first place to be honest. Seeing it now, it surely would be more intuitive put alongside the bot's response assertion instead, something like;

```diff
-  t.tearDown(() => prCommitsScope.done() && scope.done())

  supertest(app)
    .post('/node/jenkins/start')
    .send(jenkinsPayload)
    .expect(201)
    .end((err, res) => {
+      prCommitsScope.done()
+      scope.done()
       t.equal(err, null)
    })
```

Is that along the lines of what you had in mind too @refack?

Reply:
Yeah, that _looks_ more "idiomatic" BDD ¯\\_(ツ)_/¯